File: 3042_-_Count_Prefix_and_Suffix/3042.cpp
class Solution {
  public:
    int countPrefixSuffixPairs(vector<string> &words) {
        int n = words.size();
        int ans = 0;

        for (int i = 0; i < n; i++) {
            int xfix_len = words[i].size(); 
            for (int j = i + 1; j < n; j++) {
                int word_len = words[j].size();

                if (xfix_len > word_len) {
                    continue;
                }

                for (int k = 0, l = word_len - xfix_len; k < xfix_len; k++, l++) {
                    if (words[j][k] != words[i][k] || words[j][l] != words[i][k]) {
                        break;
                    }

                    if (k == xfix_len - 1) {
                        ans++;
                    }
                }
            }
        }

        return ans;
    }
};

File: 3042_-_Count_Prefix_and_Suffix/3042.py
class Solution:
    def countPrefixSuffixPairs(self, words: List[str]) -> int:
        words_len = len(words)
        results = 0

        for i in range(words_len):
            for j in range(i + 1, words_len):
                if words[j].startswith(words[i]) and words[j].endswith(words[i]):
                    results += 1

        return results
